src/components/Board.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react'
+import { determineWin } from '../utils/determineWin'
+import { Moves } from './Moves'
+import { Square } from './Square'
+import { Status } from './Status'
+
+export function Board(): JSX.Element {
+  const [turn, setTurn] = useState(0)
+  const [boardState, setBoardState] = useState([
+    {
+      board: ['', '', '', '', '', '', '', '', ''],
+    },
+  ])
+  const [playerOne, setPlayerOne] = useState(true)
+
+  /*
+    access boardState to record player's moves and save them into array;
+    this accesses the board property within boardState for each turn
+  */
+  const currentBoardState = boardState.slice(0, turn + 1)
+  const currentMove = currentBoardState[currentBoardState.length - 1]
+  const board = currentMove.board.slice(0)
+  const winner = determineWin(board)
+
+  function handleTurn(index: number) {
+    if (determineWin(board)) {
+      return board[index]
+    }
+
+    setTurn(turn + 1)
+
+    playerOne ? (board[index] = 'X') : (board[index] = 'O')
+
+    playerOne ? setPlayerOne(false) : setPlayerOne(true)
+
+    setBoardState([...boardState, { board: board }])
```

Comment:
Very clean! just a few minor syntax touchups you can do. 

Also I think this is what Isaiah meant by wanting this to be a pure function. You are mutating a variable (`board`) that is declared outside the scope of the function. While this code works trust me when I say that this pattern leads to a world of hurt and pain as it can cause really weird state bugs.

To skirt around that just declare board as an internal variable of the function.

PS `[...arr]` does the same thing as `.slice(0)` in doing a copy by value instead of by reference.
```suggestion
    const board = [...boardState[boardState.length - 1].board]
    
    setTurn(turn + 1)

    board[index] = playerOne ? 'X' : 'O'

    setPlayerOne(!playerOne)

    setBoardState([...boardState, { board }])
```